src/aks-preview/azext_aks_preview/agentpool_decorator.py

```diff
@@ -86,7 +86,17 @@ def get_vm_set_type(self) -> str:
         :return: string
         """
         # read the original value passed by the command
-        vm_set_type = self.raw_param.get("vm_set_type", CONST_VIRTUAL_MACHINE_SCALE_SETS)
+        vm_set_type = self.raw_param.get("vm_set_type")
+        if vm_set_type is None:
+            if self.raw_param.get("vm_sizes") is None:
```

Comment:
Is it possible to replace `self.raw_param.get("vm_sizes")` with `self.get_vm_sizes()`?

Reply:
`self.get_vm_sizes()` will return default value if the param is not given. Here we only want to check user's input.